signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
> That definitely needs to happen somewhere, and right now it doesn’t. This is a bad fit, though, we should be verifying the user-specified digest even if no signatures exist or are required. So far I was thinking that the individual ImageSources which support digest references should do that in their GetManifests; it could probably be shared in copy.Image, OTOH it seems reasonable that this should be rejected for any use of that ImageSource.

isn't this a separate issue though? I agree we need to verify the user provided digest against what we retrieve from the network though.

I was more talking about this new `UntaggedReference` match. So the default will be `matchAnyTag` which means an user is trying to pull `testimage@sha256:HEX_1` but the user has only signed `testimage:latest` and latest's manifest happens to be `sha256:HEX_1`.
Question, how do we handle this for `matchAnyTag`? The signature is valid for the tag and the user requests the digest for that tag so we allow that pull since the tag's manifest digest == user's pull reference digest. What happen otherwise? Shall we error out? Anyhow, I think  this confusion come from `matchAnyTag` since it will match only the tag whose manifest's digest == user's pull reference digsest and not really _any tag_ for an untagged reference.

Also, why do we call it _UntaggedReference_? an user can't initiate a pull w/o a valid tag or reference. We should call it _CanonicalReference_ maybe. The pull all tags case in docker isn't different since what it does underneath is to append every tag to the image name and download the image:tag (hence it's not _untagged_)